examples/connected-components/Makefile

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, The Pallene Developers
+# Pallene Tracer is licensed under the MIT license.
+# Please refer to the LICENSE and AUTHORS files for details
+# SPDX-License-Identifier: MIT
+
+OUTPUT_FILE = component.so
```

Comment:
Can we have a single makefile for all the examples, instead of copy-pasted makefiles for everyone? These makefiles are too complicated to be copy-pasted.

Reply:
If we did/could, what was the point of detaching them from the root Makefile? I don't find them complicated, atleast compared to the root Makefile that we have. A moderate Makefile is far better than having no Makefile at all.

Reply:
The main point of detaching them from the root is because we want to compile the examples after we do `make install` for the library, because that way it tests that the compilation will also work for the users.

I'd be ok copy-pasting trivial makefiles but these makefiles are not trivial. For starters:

1. Long list of cflags
2. Non-portable ifneq, MAKECMDGOALS
3. Multiple phony targets
4. Pattern rules (despite building a single file)

Reply:
That was not the main point actually, you can compile examples after installation from root Makefile as well, it's not a big deal. The main point was, as Pallene Tracer have 2 modes (debug and release controlled by `PT_DEBUG` macro), we have to show users the techniques to use these modes in a Makefile.

I can remember you telling me Makefiles need some love? :3

Reply:
- Using makefile variables is more natural than if statements
- I think we should keep the example makefiles as small as possible. For example, we can probably get rid of CFLAGS, the pattern rules, and much more.